modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -541,6 +541,38 @@ def reindex(self, axis, labels, **kwargs):
         return self.__constructor__(new_modin_frame)
 
     def reset_index(self, **kwargs):
+        if self.lazy_execution:
+
+            def _reset(df, *axis_lengths, partition_idx):
+                df = df.reset_index(**kwargs)
+
+                if isinstance(df.index, pandas.RangeIndex):
+                    # If the resulting index is a pure RangeIndex that means that
+                    # `.reset_index` actually dropped all of the levels of the
+                    # original index and so we have to recompute it manually for each partition
+                    start = sum(axis_lengths[:partition_idx])
+                    stop = sum(axis_lengths[: partition_idx + 1])
+
+                    df.index = pandas.RangeIndex(start, stop)
+                return df
+
+            if self._modin_frame._columns_cache is not None and kwargs["drop"]:
+                new_columns = self._modin_frame._columns_cache
+            else:
+                new_columns = None
+
+            return self.__constructor__(
+                self._modin_frame.broadcast_apply_full_axis(
```

Comment:
```suggestion
                self._modin_frame.apply_full_axis(
```
Change this and propagate the new added parameters to both `apply_full_axis` and `broadcast_apply_full_axis`. I think all the parameters should be in sync for the two methods. Or we could leave only one of the methods.

Reply:
done

Reply:
Create an issue to sync params for both methods please. I think it is bit confusing to call broadcast_apply_full_axis and pass None for other.

Reply:
https://github.com/modin-project/modin/issues/5620